### PR TITLE
4422 gzip errors

### PR DESCRIPTION
**Pull request by gabdank**

### CODE REVIEW

File: checkfiles/checkfiles.py [hunk 1]
```diff
@@ -183,6 +185,258 @@ def check_format(encValData, job, path):
...
+    return set_to_return
+
+def process_read_lengths(read_lengths_dict,
+                         lengths_list,
+                         submitted_read_length,
+                         read_count,
+                         threshold_percentage,
+                         errors_to_report,
+                         result):
+    expected_read_lengths = [
```
Comment (hitz) [inline checkfiles/checkfiles.py]: this whole bit is pretty clunky too.
you have a  histogram of read lengths and a submitted_length +/- 2.

You want something like a dictionary slice of read_lengths_dict for the 5 keys.
[ number for length, number in read_length_dict.items() if submitted_length -2 <= length <= submitted_length+2 ]
and then sum that array.

**Comment by gabdank** [comment]
Aditi currently runs it from the branch
Unfortunately this code is written as we hit new and unexpected fastq read names cases - making the code clunky.. 
I will work on refactoring using your guidlines

File: checkfiles/checkfiles.py [hunk 2]
```diff
@@ -183,6 +185,258 @@ def check_format(encValData, job, path):
...
+    read_numbers_set = set()
+    signatures_set = set()
+    signatures_no_barcode_set = set()
+    read_lengths_dictionary = {}
+    read_count = 0
+    old_illumina_current_prefix = 'empty'
+    try:
+        line_index = 0
+        for encoded_line in fastq_data_stream.stdout:
+            line = encoded_line.decode()
```
Comment (gabdank) [inline checkfiles/checkfiles.py]: DONE

File: checkfiles/checkfiles.py [hunk 3]
```diff
@@ -183,6 +185,258 @@ def check_format(encValData, job, path):
...
+            line_index += 1
+            if line_index == 1:
+                read_name = line.strip()
+                words_array = re.split(r'[\s]', read_name)
```
Comment (gabdank) [inline checkfiles/checkfiles.py]: DONE

File: checkfiles/checkfiles.py [hunk 4]
```diff
@@ -183,6 +185,258 @@ def check_format(encValData, job, path):
...
+                            read_number + ':')
+
+            if line_index == 2:
+                read_count += 1
+                length = len(line.strip())
+                if length not in read_lengths_dictionary:
+                    read_lengths_dictionary[length] = 0
+                read_lengths_dictionary[length] += 1
+            line_index = line_index % 4
+    except IOError:
```
Comment (gabdank) [inline checkfiles/checkfiles.py]: refactored

File: checkfiles/checkfiles.py [hunk 5]
```diff
@@ -183,6 +185,258 @@ def check_format(encValData, job, path):
...
+        else:
+            errors['not_unique_flowcell_details'] = str(conflicts.append(
+                ' Gathered information about the file was: {}.'.format(str(result))))
+
+
+def process_barcodes(signatures_set):
+    set_to_return = set()
+    flowcells_dict = {}
+    for entry in signatures_set:
+        arr = entry.split(':')
```
Comment (gabdank) [inline checkfiles/checkfiles.py]: DONE

File: checkfiles/checkfiles.py (hunk 1 above)
Comment (gabdank) [inline checkfiles/checkfiles.py]: DONE

**Pull request merged**